replica/table.cc

```diff
@@ -1010,11 +1025,13 @@ void table::update_stats_for_new_sstable(const sstables::shared_sstable& sst) no
 
 future<>
 table::do_add_sstable_and_update_cache(sstables::shared_sstable sst, sstables::offstrategy offstrategy, bool trigger_compaction) {
+    compaction_group& cg = compaction_group_for_sstable(sst);
+    auto holder = cg.async_gate().hold();
```

Comment:
This deserves a comment: why do we need to hold the gate. It sounds like a standalone fix beyond this series.